testament/categories.nim

```diff
@@ -563,13 +563,13 @@ proc `&.?`(a, b: string): string =
   # candidate for the stdlib?
   result = if b.startsWith(a): b else: a & b
 
-proc processSingleTest(r: var TResults, cat: Category, options, test: string) =
-  let test = testsDir &.? cat.string / test
-  let target = if cat.string.normalize == "js": targetJS else: targetC
-  if fileExists(test):
-    testSpec r, makeTest(test, options, cat), {target}
-  else:
-    doAssert false, test & " test does not exist"
+proc processSingleTest(r: var TResults, cat: Category, options, test: string, targets: set[TTarget], targetsSet: bool) =
+  var targets = targets
+  if not targetsSet:
+    let target = if cat.string.normalize == "js": targetJS else: targetC
+    targets = {target}
+  doAssert fileExists(test), test & " test does not exist"
```

Comment:
This is input validaton, not an assertion.

Reply:
I agree but the `doAssert` is pre-existing, look at the diff.

we can discuss how to properly fix this in subsequent PR (`quit` is not the answer as it doesn't play well with library integration; `enforce` is the most natural fit but enforce PR https://github.com/nim-lang/Nim/pull/15606 is still open)

Reply:
``quit`` **is** an answer until testament exposes an API. And only then we can add a callback for error handling.